src/hints/block_context.rs

```diff
@@ -176,6 +147,141 @@ pub fn load_deprecated_class(
     Ok(hint_extension)
 }
 
+pub const LOAD_CLASS_FACTS: &str = indoc! {r#"
+    ids.compiled_class_facts = segments.add()
+    ids.n_compiled_class_facts = len(os_input.compiled_classes)
+    vm_enter_scope({
+        'compiled_class_facts': iter(os_input.compiled_classes.items()),
+        'compiled_class_visited_pcs': os_input.compiled_class_visited_pcs,
+    })"#
+};
+pub fn load_class_facts(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let os_input = exec_scopes.get::<StarknetOsInput>("os_input")?;
+    let compiled_class_facts_ptr = vm.add_memory_segment();
+    insert_value_from_var_name("compiled_class_facts", compiled_class_facts_ptr, vm, ids_data, ap_tracking)?;
+
+    insert_value_from_var_name("n_compiled_class_facts", os_input.compiled_classes.len(), vm, ids_data, ap_tracking)?;
+
+    let compiled_class_facts: Box<dyn Any> = Box::new(os_input.compiled_classes.into_iter());
```

Comment:
Doesn't that create a `Box<Iterator>`? I'd collect the result into a vector just to protect us in case of multiple iterations. 

Edit: Saw the code below, seems like that's the behaviour intended in the Python VM as well. Correct?

Reply:
Right. It is not any more fragile than the rest of this cairo/imperative language construction.